sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptorV2.java

```diff
@@ -78,12 +79,12 @@ private Cipher getCipher(int index) throws GeneralSecurityException {
     protected Flux<ByteBuffer> encrypt(Flux<ByteBuffer> plainTextFlux) {
         Flux<ByteBuffer> encryptedTextFlux;
         BufferStagingArea stagingArea =
-            new BufferStagingArea(GCM_ENCRYPTION_REGION_LENGTH, GCM_ENCRYPTION_REGION_LENGTH);
+            new BufferStagingArea(this.gcmEncryptionRegionLength, this.gcmEncryptionRegionLength);
 
         encryptedTextFlux =
             UploadUtils.chunkSource(plainTextFlux,
                     new com.azure.storage.common.ParallelTransferOptions()
-                        .setBlockSizeLong((long) GCM_ENCRYPTION_REGION_LENGTH))
+                        .setBlockSizeLong((long) this.gcmEncryptionRegionLength))
```

Comment:
We should maybe check with @alzimmermsft on why the implementation is creating both a buffer staging area and also chunking the source on encryption region. Mainly the buffer staging area should already be handling splitting up the plain text into byte buffer fluxes that are equal to the encryption region length and I'm not sure what the chunkSource utility would add on top of it. The big thing is I want to make sure the underlying upload implementation is uploading in the appropriate block size (and not using the encryption region length for the block size).

Reply:
This seems to have been a work around to the bug we discussed offline after reading the documentation of `UploadUtils.chunkSource`. Also, side comment, there's no reason that that API needs to take `ParallelTransferOptions`, it's just using that to `getBlockSizeLong`, we can pass `Long blockSize` instead.

Reply:
Ah yeah I think it was the reuse of the `ParallelTransferOptions` that mainly raised the concern for me. Thanks for looking into it. Otherwise, I agree the only reason that I can see `ParallelTransferOptions` being used was to reuse the `setBlockSizeLong` validation.